src/scripts/contentCapture/pdfScreenshotHelper.ts

```diff
@@ -29,42 +30,40 @@ export class PdfScreenshotHelper {
 		return new Promise<PdfScreenshotResult>((resolve, reject) => {
 			let getBinaryEvent = new Log.Event.PromiseEvent(Log.Event.Label.GetBinaryRequest);
 
-			let request = new XMLHttpRequest();
-			request.open("GET", url, true);
-
-			request.responseType = "arraybuffer";
-
-			let errorCallback = (failureInfo: OneNoteApi.RequestError) => {
-				getBinaryEvent.setStatus(Log.Status.Failed);
-				failureInfo.response = failureInfo.responseHeaders = undefined;
-				getBinaryEvent.setFailureInfo(failureInfo);
-				Clipper.logger.logEvent(getBinaryEvent);
-				reject();
+			let errorCallback = (failureInfo: Promise<OneNoteApi.RequestError>) => {
+				failureInfo.then((error) => {
+					getBinaryEvent.setStatus(Log.Status.Failed);
+					error.response = error.responseHeaders = undefined;
+					getBinaryEvent.setFailureInfo(error);
+					Clipper.logger.logEvent(getBinaryEvent);
+					reject();
+				})
 			};
 
-			request.onload = () => {
-				if (request.status === 200 && request.response) {
-					let arrayBuffer = request.response;
 
+			fetch(url)
+			.then(response => {
+				if (!response.ok) {
+					errorCallback(ErrorUtils.createRequestErrorObject(response, OneNoteApi.RequestErrorType.UNEXPECTED_RESPONSE_STATUS));
+				}
+				return response.arrayBuffer();
+			})
+				.then(arrayBuffer => {
 					getBinaryEvent.setCustomProperty(Log.PropertyName.Custom.ByteLength, arrayBuffer.byteLength);
 					Clipper.logger.logEvent(getBinaryEvent);
 
-					PdfScreenshotHelper.getPdfScreenshotResult(arrayBuffer).then((pdfScreenshotResult) => {
-						pdfScreenshotResult.byteLength = arrayBuffer.byteLength;
-						resolve(pdfScreenshotResult);
-					});
-				} else {
-					errorCallback(OneNoteApi.ErrorUtils.createRequestErrorObject(request, OneNoteApi.RequestErrorType.UNEXPECTED_RESPONSE_STATUS));
-				}
-			};
-			request.ontimeout = () => {
-				errorCallback(OneNoteApi.ErrorUtils.createRequestErrorObject(request, OneNoteApi.RequestErrorType.REQUEST_TIMED_OUT));
-			};
-			request.onerror = () => {
-				errorCallback(OneNoteApi.ErrorUtils.createRequestErrorObject(request, OneNoteApi.RequestErrorType.NETWORK_ERROR));
-			};
-
-			request.send();
+					PdfScreenshotHelper.getPdfScreenshotResult(new Uint8Array(arrayBuffer))
+						.then(pdfScreenshotResult => {
+							pdfScreenshotResult.byteLength = arrayBuffer.byteLength;
+							resolve(pdfScreenshotResult);
+						})
+				})
+			.catch(() => {
```

Comment:
Should we create a request error object here as well?

Reply:
The response object will not be available in the catch block hence not able to create a request error object here